### PR TITLE
[SPARK-32310][ML][PySpark] ML params default value parity in classification, regression, clustering and fpm

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
set params default values in trait ...Params in both Scala and Python.
I will do this in two PRs. I will change classification, regression, clustering and fpm in this PR. Will change the rest in another PR.

### Why are the changes needed?
Make ML has the same default param values between estimator and its corresponding transformer, and also between Scala and Python.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Existing tests

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @srowen @viirya @zhengruifeng

**Comment by srowen** [comment]
So in theory this shouldn't change behavior, or if it does, it's fixing an incompatibility that's likely more a bug than anything right?

**Comment by huaxingao** [comment]
> So in theory this shouldn't change behavior, or if it does, it's fixing an incompatibility that's likely more a bug than anything right?

Yes

**Comment by viirya** [comment]
"classification, regression, clustering and fpm" instead of "part 1" in the title?

File: mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala [hunk 1]
```diff
@@ -68,6 +68,12 @@ private[clustering] trait BisectingKMeansParams extends Params with HasMaxIter
     "The minimum number of points (if >= 1.0) or the minimum proportion " +
       "of points (if < 1.0) of a divisible cluster.", ParamValidators.gt(0.0))
 
+
+  setDefault(
```
Comment (zhengruifeng) [inline mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala:72]: total nit: make these params in single line, like above places

**Review by zhengruifeng — COMMENTED** [review]
LGTM

File: mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala [hunk 2]
```diff
@@ -85,7 +85,6 @@ class FMClassifier @Since("3.0.0") (
    */
   @Since("3.0.0")
   def setFactorSize(value: Int): this.type = set(factorSize, value)
-  setDefault(factorSize -> 8)
```
Comment (viirya) [inline mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala:88]: Where do the default params of `FMClassifier` move?

**Review by viirya — APPROVED** [review]
Looks okay if we are sure this doesn't change behavior.

File: mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala (hunk 1 above)
Comment (huaxingao) [reply mllib/src/main/scala/org/apache/spark/ml/clustering/BisectingKMeans.scala]: Fixed. Thanks!

File: mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala (hunk 2 above)
Comment (huaxingao) [reply mllib/src/main/scala/org/apache/spark/ml/classification/FMClassifier.scala:88]: I move setDefault for these params to FactorizationMachinesParams in FMRegressor when I fixed solver last time

**Comment by huaxingao** [comment]
@srowen If it's OK with you, I will merge after test passes so I can finish the 2nd part. I will merge to both 3.0.1 and master.

**Review by srowen — APPROVED** [review]

**Comment by huaxingao** [comment]
Merged to mater. Thank you all for reviewing!

**Pull request closed without merging**